### ISSUE TITLE
compile_commands.json

### ISSUE THREAD

**Issue by bhack** [tensorflow/addons#1894]
It could be nice if we could generate/distribute `compile_commands.json` to better interact with some IDE and our c++ code.

I.e. In Bazel and also in Vscode the the official Bazel team vscode plugin we don't have in tree support to generate `compile_commands`:
https://github.com/bazelbuild/bazel/issues/258
https://github.com/bazelbuild/vscode-bazel/issues/179

I've tried some quite popular community workaround like:
https://github.com/grailbio/bazel-compilation-database

But the problem is that we are using `--action_env` for `build` when we generate our `.bazelrc`  with `python configure.py` and so these arguments not supported in other command like  e.g. `bazel query` https://github.com/bazelbuild/bazel/issues/10226. 
This is going to invalidate the official bazel vscode plugin that need to execute `query command` but also the `bazel-compilation-database` workaround cause we have problem to retrieve our env variable.

**Comment by bhack** [comment]
/cc @gunan This could be interesting to have in Tensorflow itself but I don't want to duplicate the ticket there. 
I don't know if you have an internal route on the Bazel team.

**Comment by gunan** [comment]
I am not sure what compile_commands.json file is. 
Is this file generated by bazel?
Could you provide more details?

**Comment by bhack** [comment]
@gunan compile_commands.json Is https://clang.llvm.org/docs/JSONCompilationDatabase.html and It is used by many IDEs for integrating with c++ project.
If you see I've mentioned two Bazel issues (one for Basel itself and the other one for the Bazel tram official plugin for Vscode) cause currently you need to use some external tools/workarunds) to generate/distribuire compile_commands.json with Bazel projects.

**Comment by mihaimaruseac** [comment]
This is also useful for clangd and Clang Tooling based refactorings.

**Comment by bhack** [comment]
@mihaimaruseac are you using the tool or not? :Wink: See https://github.com/tensorflow/addons/pull/1808#issuecomment-636217541

**Comment by mihaimaruseac** [comment]
`clang-tidy` is used, but on the imported code.

What I meant above is other tooling (`clang_query` and related) and directly on the open source code

**Comment by gunan** [comment]
From what I can tell, then we would need to update this file every time a new file is added, or a BUILD file is modified?

**Comment by bhack** [comment]
@gunan I've already mentioined two bazel bugs:
bazelbuild/bazel#258
bazelbuild/vscode-bazel#179

**Comment by mihaimaruseac** [comment]
I think I might be able to do it, if I really need this for a security analysis I'm running. No promises yet.

**Comment by bhack** [comment]
As Bazel team seems It Is not interested to put resources on this at upstream level many projects are using https://github.com/grailbio/bazel-compilation-database

**Comment by mihaimaruseac** [comment]
Yes, but I think I have a solution which uses Bazel and a minimal number of additions to the projects `.bazelrc`

**Comment by bhack** [comment]
Ok but check to not reinvet the wheel of alternative approaches:
 https://github.com/grailbio/bazel-compilation-database#alternatives

**Comment by mihaimaruseac** [comment]
Yes, I'm only trying to get the solution that is most stable and requires fewer changes

**Comment by bhack** [comment]
@mihaimaruseac Do you think that we could track this issue in the build SIG repo?

**Comment by mihaimaruseac** [comment]
Yes, let's do that there. Can you open an issue for me please?

I got preempted from working on this, apologies for the delay

**Comment by bhack** [comment]
Sorry I have no superpower to migrate this issue.

**Comment by mihaimaruseac** [comment]
Oh, I meant creating a new one and then we can either close this or just mention this in the other
